### PR TITLE
TG-813 Upgrade to Python 3.11; update requirements; add Ruff

### PR DESCRIPTION
**Pull request by niccolomineo**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,14 +3,12 @@
 .PHONY: check
 check:  ## Check code formatting and import sorting
 	python3 -m black --check .
-	python3 -m isort --check .
-	python3 -m flake8
+	ruff .
```
Comment (pauloxnet) [inline Makefile:6]:
```suggestion
	python3 -m ruff .
```

File: Makefile [hunk 2]
```diff
@@ -3,14 +3,12 @@
...
 
 .PHONY: fix
 fix:  ## Fix code formatting, linting and sorting imports
 	python3 -m black .
-	python3 -m isort .
-	python3 -m flake8
+	ruff --fix .
```
Comment (pauloxnet) [inline Makefile:11]:
```suggestion
	python3 -m ruff --fix .
```

File: setup.cfg [hunk 3]
```diff
@@ -1,15 +1,2 @@
...
-    .mypy_cache,
-    .pytest_cache,
-    .venv,
-    .vscode*,
-ignore = E501,I003,T499,W503
-max-line-length = 80
-mypy_config = setup.cfg
-select = B,C,D,E,F,I,W,T4,B9
-
 [metadata]
```
Comment (pauloxnet) [inline setup.cfg:1]: Remove this file

**Review by pauloxnet — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (niccolomineo) [reply Makefile:11]: Non utilizzabile, purtroppo.

File: Makefile (hunk 1 above)
Comment (niccolomineo) [reply Makefile:6]: Non utilizzabile, purtroppo.

File: setup.cfg (hunk 3 above)
Comment (niccolomineo) [reply setup.cfg]: Ok

**Review by Mitchina — COMMENTED** [review]
It seems okay to me, I just can't test it

File: pyproject.toml [hunk 4]
```diff
@@ -14,5 +14,38 @@ known_third_party = [
...
+    "D415",
+    "D416",
+    "D417",
+    "E501",
+ ]
+ select = ["B", "C", "D", "E", "F", "I", "W", "B9"]
+ target-version = "py311"
+
+ [tool.ruff.isort]
+ known-first-party = [
```
Comment (pauloxnet) [inline pyproject.toml:49]: Copy here setting from `isort` section and remove it.

**Review by pauloxnet — CHANGES_REQUESTED** [review]

**Review by pauloxnet — APPROVED** [review]

**Pull request merged**